lib/config/validation.ts

```diff
@@ -634,6 +634,26 @@ export async function validateConfig(
                   message: `Invalid \`${currentPath}.${key}.${res}\` configuration: value is not a string`,
                 });
               }
+            } else if (key === 'encrypted') {
+              const res = validatePlainObject(val);
+              if (res !== true) {
+                errors.push({
+                  topic: 'Configuration Error',
+                  message: `Invalid \`${currentPath}.${key}.${res}\` configuration: value is not a string`,
+                });
+              }
+
+              const privateKey =
+                configType === 'global'
+                  ? (config.privateKey ?? config.privateKeyOld)
```

Comment:
When decrypting, does renovate try privateKeyOld if no privateKey is found?

Reply:
private key should be required if private key old is defined.

Reply:
We need to model the current behavior